cmd/dlt/accountroles/cmd.go

```diff
@@ -0,0 +1,254 @@
+		"role-name",
+		"r",
+		"",
+		"Account role name to be deleted.",
+	)
+
+	flags.StringVarP(
+		&args.prefix,
+		"prefix",
+		"p",
+		"",
+		"Prefix of the account roles to be deleted.",
+	)
+
+	confirm.AddFlag(flags)
+}
+
+func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return modes, cobra.ShellCompDirectiveDefault
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	if cmd.Flags().Changed("prefix") && cmd.Flags().Changed("role-name") {
+		reporter.Errorf("Provide either prefix or role-name not both")
+		os.Exit(1)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	env, err := ocm.GetEnv()
+	if err != nil {
+		reporter.Errorf("Error getting environment %s", err)
+		os.Exit(1)
+	}
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+	clusters, err := ocmClient.GetClusters(creator, 1000)
+	if err != nil {
+		reporter.Errorf("Error getting clusters %s", err)
+		os.Exit(1)
+	}
+	finalRoleList := []string{}
+	if args.roleName != "" {
+		role, err := awsClient.GetAccountRoleForCurrentEnv(env, args.roleName)
+		if err != nil {
+			reporter.Errorf("Error getting role: %s", err)
+			os.Exit(1)
+		}
+		if !checkIfRoleExists(clusters, role) {
+			finalRoleList = append(finalRoleList, role.RoleName)
+		}
+	} else if args.prefix != "" {
+		roles, err := awsClient.GetAccountRoleForCurrentEnvWithPrefix(env, args.prefix)
+		if err != nil {
+			reporter.Errorf("Error getting role: %s", err)
+			os.Exit(1)
+		}
+		for _, role := range roles {
+			if !checkIfRoleExists(clusters, role) {
+				finalRoleList = append(finalRoleList, role.RoleName)
+			}
+		}
+		if len(finalRoleList) == 0 {
+			reporter.Errorf("There are no roles to be deleted")
+			os.Exit(1)
+		}
+	} else {
+		roles, err := awsClient.GetAccountRolesForCurrentEnv(env, creator.AccountID)
+		if err != nil {
+			reporter.Errorf("Error getting roles: %s", err)
+			os.Exit(1)
+		}
+		for _, role := range roles {
+			if !checkIfRoleExists(clusters, role) {
+				finalRoleList = append(finalRoleList, role.RoleName)
+			}
+		}
+		if len(finalRoleList) == 0 {
+			reporter.Errorf("There are no roles to be deleted")
+			os.Exit(1)
+		}
```

Comment:
This block is repeated in all the `if` conditions above, maybe put it just belog the closing bracked so that it applies to all three.